ONI-DenseLogic/LogicSevenSegment.cs

```diff
@@ -0,0 +1,85 @@
+ * distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+using KSerialization;
+using UnityEngine;
+
+namespace ONI_DenseLogic {
+	[SerializationConfig(MemberSerialization.OptIn)]
+	public class LogicSevenSegment : KMonoBehaviour, IRender200ms {
+		public static readonly HashedString INPUTID = new HashedString("LogicSevenSegment_IN");
+		public static readonly HashedString OUTPUTID = new HashedString("LogicSevenSegment_OUT");
+
+		public static readonly CellOffset INPUTOFFSET = new CellOffset(0, 0);
+		public static readonly CellOffset OUTPUTOFFSET = new CellOffset(0, 2);
+
+		private static readonly EventSystem.IntraObjectHandler<LogicSevenSegment>
+		OnLogicValueChangedDelegate = new EventSystem.IntraObjectHandler<LogicSevenSegment>(
+		(component, data) => component.OnLogicValueChanged(data));
+
+#pragma warning disable IDE0044 // Add readonly modifier
+		[MyCmpReq]
+		private KBatchedAnimController kbac;
+#pragma warning restore IDE0044
+
+		[Serialize]
+		private int inVal;
+		private int curOut;
+		private MeterController meter;
+
+		protected override void OnSpawn() {
+			base.OnSpawn();
+			Subscribe((int)GameHashes.LogicEvent, OnLogicValueChangedDelegate);
+			meter = new MeterController(kbac, "meter_target", kbac.FlipY ? "meter_dn" : "meter_up", Meter.Offset.UserSpecified, Grid.SceneLayer.LogicGatesFront, Vector3.zero, null);
+			UpdateMeter();
+		}
+
+		public void UpdateMeter() {
+			meter.SetPositionPercent((inVal % 10) / 10f);
+		}
+
+		public void OnLogicValueChanged(object data) {
+			var logicValueChanged = (LogicValueChanged)data;
+			if (logicValueChanged.portID == INPUTID)
+				inVal = logicValueChanged.newValue;
+			else
+				return;
+			UpdateMeter();
+			UpdateLogicCircuit();
+		}
+
+		private void UpdateLogicCircuit() {
+			curOut = inVal > 9 ? 1 : 0;
+			GetComponent<LogicPorts>().SendSignal(OUTPUTID, curOut);
+			UpdateVisuals();
+		}
+
+		public void Render200ms(float dt) {
+			// hexi/test/peter: Do we have to do this here? Can we render only on state change?
+			UpdateMeter();
+			UpdateVisuals();
+		}
+
+		public void UpdateVisuals() {
+			int num0 = inVal > 0 ? 1 : 0;
+			int num1 = curOut;
+			int state = num0 + 4 * num1;
```

Comment:
Should this not be "4 * num0 + num1"?

Also, "inVal > 0 ? 1 : 0" could be merged into this statement as "inVal > 0 ? 4 + num1 : num1" or "1 + num1 * 4 : num1 * 4" depending on where the multiplication is supposed to go.

Reply:
I did it this way to be consistent with the way I've written out the `on_{number}` calculations for the other things. I suppose I could change all of them but I think I'll leave this as it. I think it reads pretty clearly.